activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -565,7 +569,7 @@ def table_options(table_name) # :nodoc:
 
       # SHOW VARIABLES LIKE 'name'
       def show_variable(name)
-        query_value("SELECT @@#{name}", "SCHEMA")
+        query_value("SELECT @@#{name}", "SCHEMA", materialize_transactions: false, allow_retry: true)
```

Comment:
I actually wonder if `query_value` shouldn't default to `materialize_transactions: false, allow_retry: true`, I'd need to review all of usages, but it's normally mean to read connection variables and schema, and using `query_value` to do a write would be weird...